snakemake/__init__.py

```diff
@@ -2376,6 +2408,11 @@ def adjust_path(f):
                     file=sys.stderr,
                 )
                 sys.exit(1)
+    if args.slurm:
+        args.use_envmodules = True
+        if args.cores is None:
+            args.cores = 1  # TODO: check - is this ok?
+
```

Comment:
```suggestion
```

Not needed altogether:
1. I would not imply envmodules on slurm. There are certainly people who don't use them. Such defaults are a job for a corresponding profile.
2. Cores should not be set here, Snakemake has logic for handling cores = None at a later stage.

Reply:
ad 1: Yes, admitted. On smaller cluster that will be true.
ad 2: OK.